Show # genes matched as integer in PatientView

msalihaltun opened the pull request:
Address point 3 in https://github.com/cBioPortal/cbioportal/issues/8078#issuecomment-732722936.

Change `package.json` to point to latest commit in PathwayMapper where point 1 in above comment is addressed.

msalihaltun committed:
Show # genes matched as integer in PatientView

  In patient view table, show the content of # genes matched column
  as integer instead of float. Add an optional property to
  PathwayMapperTable component where the score representation
  can be converted into any string representation with a function.

onursumer commented on src/pages/patientView/pathwayMapper/PatientViewPathwayMapper.tsx:
```diff
@@ -170,6 +170,7 @@ export default class PatientViewPathwayMapper extends React.Component<
                         tooltip: <span>Number of Genes Matched</span>,
                     },
```
Let's just define a custom render function here for the score column. This way, you wouldn't need to modify the default render function inside `PathwayMapperTable`, and you wouldn't need to define another prop like `convertScoreRepresentation`.

```suggestion
                        render: (d: IPathwayMapperTable) => (
                            <span>
                                <b>{d.score.toFixed(0)}</b>
                           </span>
                       ),
                    },
```

msalihaltun replied:
I didn't know I could do that :)

onursumer replied:
no worries :)

onursumer commented on src/shared/lib/pathwayMapper/PathwayMapperTable.tsx:
```diff
@@ -133,11 +134,18 @@ export default class PathwayMapperTable extends React.Component<
-                    <b>{d.score.toFixed(2)}</b>
-                </span>
-            ),
+            render: (d: IPathwayMapperTable) => {
```
Let's just leave the default render as is. See my comments about an alternative solution.

onursumer commented on src/shared/lib/pathwayMapper/PathwayMapperTable.tsx:
```diff
@@ -34,6 +34,7 @@ interface IPathwayMapperTableProps {
     columnsOverride?: {
         [columnEnum: number]: Partial<PathwayMapperTableColumn>;
     };
+    convertScoreRepresentation?: (score: number) => string;
```
Let's remove this prop.

onursumer reviewed (changes requested):
Let's just simplify this PR by defining a custom render function for the score column.

msalihaltun committed:
Use custom render function to convert score.

  Instead of a new prop to convert how score is represented, use a
  custom render function inside columnsOverride.
